templates/java/org/prism/Nodes.java.erb

```diff
@@ -92,13 +92,17 @@ public abstract class Nodes {
 
         public static final Node[] EMPTY_ARRAY = {};
 
+        public final int nodeId;
```

Comment:
I think for now it would be best to not include this because it adds 8 or 0 bytes (due to 64-bit word alignment) for each Node, and AFAIK currently it has no purpose in Java.
Specifically (AFAIK) there is no public Ruby API to access the `node_id` or use it, only RubyVM stuff, which is CRuby-specific.

---

I feel the node_id stuff in CRuby was done in a rather rushed way, I think it would be worth to gather the requirements, design a proper API for this in Prism, and clarify what information the Ruby implementations need to keep and provide.
For instance TruffleRuby already keeps the startOffset & length of every node, isn't that already enough to identify a Node?